12.LOCO_BULK_MLP/0.data_splits/scripts/0.LOCO_data_split.py

```diff
@@ -0,0 +1,208 @@
+    gsdmd_features
+) + len(mito_features) + len(er_features) + len(other_features) + len(
+    correlation_features
+) + len(
+    metadata_features
+)
+
+
+# In[6]:
+
+
+# create the combinations of features
+# out of 5 channels, we can have 0, 1, 2, 3, 4, or 5 channels
+# even with 0 channels, we still have the non channel features
+# these are areshape features
+# set the feature combination lists
+dict_of_feature_combinations = {
+    "No_channels": other_features,
+    "CorrDNA": nuclei_features,
+    "CorrPM": pm_features,
+    "CorrGasdermin": gsdmd_features,
+    "CorrMito": mito_features,
+    "CorrER": er_features,
+    "CorrDNA_CorrPM": nuclei_features + pm_features,
+    "CorrDNA_CorrGasdermin": nuclei_features + gsdmd_features,
+    "CorrDNA_CorrMito": nuclei_features + mito_features,
+    "CorrDNA_CorrER": nuclei_features + er_features,
+    "CorrPM_CorrGasdermin": pm_features + gsdmd_features,
+    "CorrPM_CorrMito": pm_features + mito_features,
+    "CorrPM_CorrER": pm_features + er_features,
+    "CorrGasdermin_CorrMito": gsdmd_features + mito_features,
+    "CorrGasdermin_CorrER": gsdmd_features + er_features,
+    "CorrMito_CorrER": mito_features + er_features,
+    "CorrDNA_CorrPM_CorrGasdermin": nuclei_features + pm_features + gsdmd_features,
+    "CorrDNA_CorrPM_CorrMito": nuclei_features + pm_features + mito_features,
+    "CorrDNA_CorrPM_CorrER": nuclei_features + pm_features + er_features,
+    "CorrDNA_CorrGasdermin_CorrMito": nuclei_features + gsdmd_features + mito_features,
+    "CorrDNA_CorrGasdermin_CorrER": nuclei_features + gsdmd_features + er_features,
+    "CorrDNA_CorrMito_CorrER": nuclei_features + mito_features + er_features,
+    "CorrPM_CorrGasdermin_CorrMito": pm_features + gsdmd_features + mito_features,
+    "CorrPM_CorrGasdermin_CorrER": pm_features + gsdmd_features + er_features,
+    "CorrPM_CorrMito_CorrER": pm_features + mito_features + er_features,
+    "CorrGasdermin_CorrMito_CorrER": gsdmd_features + mito_features + er_features,
+    "CorrDNA_CorrPM_CorrGasdermin_CorrMito": nuclei_features
+    + pm_features
+    + gsdmd_features
+    + mito_features,
+    "CorrDNA_CorrPM_CorrGasdermin_CorrER": nuclei_features
+    + pm_features
+    + gsdmd_features
+    + er_features,
+    "CorrDNA_CorrPM_CorrMito_CorrER": nuclei_features
+    + pm_features
+    + mito_features
+    + er_features,
+    "CorrDNA_CorrGasdermin_CorrMito_CorrER": nuclei_features
+    + gsdmd_features
+    + mito_features
+    + er_features,
+    "CorrPM_CorrGasdermin_CorrMito_CorrER": pm_features
+    + gsdmd_features
+    + mito_features
+    + er_features,
+    "All_channels": nuclei_features
+    + pm_features
+    + gsdmd_features
+    + mito_features
+    + er_features
+    + other_features
+    + correlation_features,
+}
+
+for combination in dict_of_feature_combinations:
+    if combination == "No_channels":
+        temp_correlation_features = other_features
```

Comment:
I am a bit confused here, why would the AreaShape feature be added into this correlation features list?

Reply:
Area shape is not a channel specific feature per se